packages/changed-elements-react/src/widgets/V2Widget/hooks/useNamedVersionLoader.tsx

```diff
@@ -0,0 +1,354 @@
+  setResult: (result: NamedVersionLoaderResult) => void;
+  isDisposed: () => boolean;
+};
+
+type Entry = {
+  version: NamedVersion;
+  state: VersionProcessedState;
+  jobStatus: JobStatus;
+  jobProgress: JobProgress;
+};
+
+const processChangesetsAndUpdateResultState = async (args: ProcessChangesetsArgs) => {
+  const currentVersionId = args.namedVersionLoaderState.result.namedVersions.currentVersion.version.id;
+
+  // deletes all comparison jobs only for debugging purposes
+  // await Promise.all(args.namedVersionLoaderState.result.namedVersions.entries.map(async (entry) => {
+  //await deleteJob(args.comparisonJobClient, entry, args.iTwinId, args.iModelId, currentVersionId);
+  // })).catch(() => { return; });
+  const newEntries = await Promise.all(args.namedVersionLoaderState.result.namedVersions.entries.map(async (entry) => {
+    const jobStatusAndJobProgress: JobStatusAndJobProgress = await getJobStatusAndJobProgress(args.comparisonJobClient, entry, args.iTwinId, args.iModelId, currentVersionId);
+    return {
+      version: entry.version,
+      state: VersionProcessedState.Processed,
+      jobStatus: jobStatusAndJobProgress.jobStatus,
+      jobProgress: jobStatusAndJobProgress.jobProgress,
+    };
+  }));
+  args.namedVersionLoaderState.result = {
+    namedVersions: { currentVersion: args.namedVersionLoaderState.result.namedVersions.currentVersion, entries: newEntries },
+  };
+  args.setResult(args.namedVersionLoaderState.result);
+  void pollForProgressingJobs({
+    iTwinId: args.iTwinId,
+    iModelId: args.iModelId,
+    namedVersionLoaderState: args.namedVersionLoaderState,
+    comparisonJobClient: args.comparisonJobClient,
+    setResult: args.setResult,
+    isDisposed: args.isDisposed,
+  });
+};
+
+type UpdateProgressingChangesetsArgs = {
+  iTwinId: string;
+  iModelId: string;
+  namedVersionLoaderState: NamedVersionLoaderState;
+  comparisonJobClient: IComparisonJobClient;
+  setResult: (result: NamedVersionLoaderResult) => void;
+  isDisposed: () => boolean;
+};
+const pollForProgressingJobs = async (args: UpdateProgressingChangesetsArgs) => {
+  if (args.isDisposed())
+    return;
+  const currentVersionId = args.namedVersionLoaderState.result.namedVersions.currentVersion.version.id;
+  let entries = args.namedVersionLoaderState.result.namedVersions.entries.slice();
+  const areThereProgressingJob = (entries: Entry[]) => {
+    return entries.find(entry => entry.jobStatus === "Processing" || entry.jobStatus === "Queued") !== undefined;
+  };
+  if (areThereProgressingJob(entries)) {
+    const idEntryMap = new Map<string, Entry>();
+    entries.forEach((entry) => idEntryMap.set(entry.version.id, entry));
+    let updatingEntries = entries.filter((entry) => entry.jobStatus === "Processing" || entry.jobStatus === "Queued");
+    while (updatingEntries.length > 0 && !args.isDisposed()) {
+      for (let entry of updatingEntries) {
+        await new Promise((resolve) => setTimeout(resolve, 5000));
```

Comment:
Why is this 5 second delay here? Can you remove it or add a comment indicating why it is here/necessary?